azurelinuxagent/common/osutil/ubuntu.py

```diff
@@ -132,6 +134,25 @@ def start_agent_service(self):
     def stop_agent_service(self):
         return shellutil.run("systemctl stop {0}".format(self.service_name), chk_err=False)
 
+    def get_dhcp_lease_endpoint(self):
+        pathglob = "/run/systemd/netif/leases/*"
+        logger.info("looking for leases in path [{0}]".format(pathglob))
+        endpoint = None
+        for lease_file in glob.glob(pathglob):
+            with open(lease_file) as f:
+                lease = f.read()
+            for line in lease.splitlines():
+                if line.startswith("OPTION_245"):
+                    option_245 = line.split("=")[1]
+                    options = [int(i, 16) for i in textwrap.wrap(option_245, 2)]
+                    endpoint = "{0}.{1}.{2}.{3}".format(*options)
```

Comment:
please ensure the data is expected, or catch IndexError (incorrect # of octects) & ValueError (non-hex value) exceptions and fall back to the default

Reply:
I can just catch general exceptions since there are already defaults in [dhcp](https://github.com/Azure/WALinuxAgent/blob/28345a55f9b21dae89472111635fd6e41809d958/azurelinuxagent/common/dhcp.py#L53)